### ISSUE TITLE
Unique page name should go before site or book title for HTML output [a11y]

### ISSUE THREAD

**Issue by batpigandme** [quarto-dev/quarto-cli#8904]
### Bug description

Currently, if HTML output is part of a website or book, the page title is formatted such that the site name or book title precedes the page name or chapter title. This is contrary to WCAG standards for [informative, unique page titles](https://www.w3.org/WAI/tips/writing/#provide-informative-unique-page-titles), where the gist of it is that it should go page title, then site/book/org name (here's [WCAG Understanding SC 2.4.2](https://www.w3.org/WAI/WCAG21/Understanding/page-titled), but I think I've summed it up pretty well).

For HTML output, I believe it stems from the template line here:
https://github.com/quarto-dev/quarto-cli/blob/fa49843c9920991f301db0aab00d338e5a51a880/src/resources/formats/html/pandoc/html.template#L19

or possibly in the metadata partial:
https://github.com/quarto-dev/quarto-cli/blob/fa49843c9920991f301db0aab00d338e5a51a880/src/resources/formats/html/pandoc/metadata.html#L23

### Steps to reproduce

All Quarto pages work this way right now. For example, from the Quarto docs: the [Get Started](https://quarto.org/docs/get-started/) page (from https://github.com/quarto-dev/quarto-web/blob/main/docs/get-started/index.qmd) has the following title element:
```html
<title>Quarto – Get Started</title>
```

Better output would be:

```html
<title>Get Started – Quarto</title>
```

### Expected behavior

HTML output should have:
```html
<title>unique page name – site name</title>
```

If you edit the metadata.html partial, this is accomplished with:
```html
<title>$pagetitle$ – $if(title-prefix)$$title-prefix$$endif$</title>
```

### Actual behavior

Super simplified versions from current HTML header:
```html
<head>
<title>sitename – page name</title>
</head>
```

Or, from a book (where 1 represents whatever chapter number it is):
```html
<head>
<title>Book Name – 1&nbsp; Chapter Name</title>
</head>
```

Thanks for your help, and continued commitment to accessible defaults!

**Comment by cscheid** [comment]
Thanks! We should improve this. But this will look bad if `title-prefix` is empty.

```
<title>$pagetitle$ – $if(title-prefix)$$title-prefix$$endif$</title>
```

So we need to be a bit more careful.

**Comment by cderv** [comment]
Just heads up that this will have an impact on anyone using `title-prefix` variable in their configuration. Currently, this really aims to be a prefix, and we inherit from Pandoc. 

I believe problem here is about our usage of `title-prefix` to set the website name. 

Let me details this below 

## About `title-prefix` change in template

`title-prefix` is really documented as a prefix by Pandoc and so by us as we inherit from Pandoc 
* https://quarto.org/docs/reference/formats/html.html#metadata
* https://pandoc.org/MANUAL.html#option--title-prefix

This change in HTML template will apply also to single document using `title-prefix` for some reason. 

For website, this is available to use it since https://github.com/quarto-dev/quarto-cli/issues/5196 and it follows currently what Pandoc's offer as configuration using [`--title-prefix`](https://pandoc.org/MANUAL.html#option--title-prefix) 

If someone really uses it to set a prefix, this will revert the order the user, and it won't be anymore.

If we really want to make this  it we really revert the order and this is no more a prefix in Quarto
* We should probably give the same feedback to Pandoc if we think template is wrong
* We should update our documentation as this is no more a prefix
* We should probably update other places where we do them the same order (like for open graph data `og:title` which is currently built the old way) 
https://github.com/quarto-dev/quarto-cli/blob/98b7fec8d3daf0c549d91785d4afa104eb899748/src/project/types/website/website-meta.ts#L92-L102
https://github.com/quarto-dev/quarto-cli/blob/98b7fec8d3daf0c549d91785d4afa104eb899748/src/project/types/website/website-meta.ts#L298-L302
https://github.com/quarto-dev/quarto-cli/blob/98b7fec8d3daf0c549d91785d4afa104eb899748/src/project/types/website/website-shared.ts#L84-L109

## About our usage

It happens we use `title-prefix` to pass the website name 
https://github.com/quarto-dev/quarto-cli/blob/98b7fec8d3daf0c549d91785d4afa104eb899748/src/project/types/website/website.ts#L164-L170

If this is not right (which I understand this issue is about), maybe we should just append to the `pagetitle` variable, or add a `title-suffix` in template for Quarto so that we can insert this content in the right place. 

I just want to avoid any side effect for all user by this change that seems small, especially because we derive from Pandoc usage and doc now.

**Comment by mcanouil** [comment]
FWIW, I agree with Carlos here.

**Comment by cderv** [comment]
To be clear, I am ok with the change. So we all agree. So let's keep it. 
This is breaking change for someone who wanted has set a custom title-prefix to be before the page title, but this is valid and we can adjust back if anyone complains. 

However, I think we should adjust the remaining part that relates to this change

- Document correctly `title-prefix` because it is now different than the one pandoc uses, and we currently inherit the documentation. 
- Adapt our opengraph metadata to use the same order. the title should match. 

Sorry if this wasn't clear. I tend to often share two sides of the problem with all info for us to decide.

**Comment by mcanouil** [comment]
Could we do as with `self-contained`/`embed-resources` for the `title-prefix`?
